ml-algorithms/src/main/java/org/opensearch/ml/engine/MLEngine.java

```diff
@@ -34,13 +37,18 @@ public MLEngine(Path opensearchDataFolder) {
         djlModelsCachePath = djlCachePath.resolve("models_cache");
     }
 
-    public String getCIPrebuiltModelConfigPath(String modelName, String version) {
-        return String.format("https://ci.opensearch.org/ci/dbc/models/ml-models/%s/%s/config.json", modelName, version, Locale.ROOT);
+    public String getPrebuiltModelConfigPath(String modelName, String version, MLModelFormat modelFormat) {
+        String format = modelFormat.name().toLowerCase(Locale.ROOT);
+        return String.format("%s/%s/%s/%s/config.json", MODEL_REPO, modelName, version, format, Locale.ROOT);
     }
 
-    public String getCIPrebuiltModelPath(String modelName, String version) {
-        int index = modelName.lastIndexOf("/") + 1;
-        return String.format("https://ci.opensearch.org/ci/dbc/models/ml-models/%s/%s/%s.zip", modelName, version, modelName.substring(index), Locale.ROOT);
+    public String getPrebuiltModelPath(String modelName, String version, MLModelFormat modelFormat) {
+        int index = modelName.indexOf("/") + 1;
+        // /huggingface/sentence-transformers/msmarco-distilbert-base-tas-b/1.0.0/onnx/sentence-transformers_msmarco-distilbert-base-tas-b-1.0.0-torch_script.zip
+        // /huggingface/sentence-transformers/msmarco-distilbert-base-tas-b/1.0.0/onnx/config.json
+        String format = modelFormat.name().toLowerCase(Locale.ROOT);
+        String modelZipFileName = modelName.substring(index).replace("/", "_") + "-" + version + "-" + format;
+        return String.format("%s/%s/%s/%s/%s.zip", MODEL_REPO, modelName, version, format, modelZipFileName, Locale.ROOT);
```

Comment:
Same here, may be we can check if the file is actually there? if not raise an exception saying invalid file path?

Reply:
The download function will throw exception if file doesn't exist.